src/commands.ts

```diff
@@ -512,12 +513,30 @@ export const insertInlineFootnote =
         footnotesSection.node,
         schema.nodes.footnotes_element
       )
-      // TODO: Revisit this position calculation as it doesn't sound right to always push the note to the end.
+      // Get position for inserting the footnote
+      let index = 0
+      // TODO:: replace zero with body position
+      state.doc.slice(0, insertedAt).content.descendants((node) => {
+        if (node.type === schema.nodes.inline_footnote) {
+          index++
+          return false
+        }
+      })
+
+      let correctPosition = 0
+      if (index != 0) {
+        footnoteElement[0].node.descendants((node, pos, parent, nodeIndex) => {
+          if (node.type === schema.nodes.footnote) {
+            if (index > nodeIndex) {
```

Comment:
Hmmm, this confuses me. How did you cover the < or = cases?

Reply:
if index is equal to or less than nodeIndex, the position was calculated in previous steps of loop

Reply:
Please move this position calculation logic to footnotes-utils and call it `getNewFootnotePos`. We will need this to be refined there so we can use it with the table footnotes too.

Reply:
I suggested something like:

```
function getNewFootnotePos(
  footnotesElementAndPos: NodeWithPos ,
  label: number
) {
  // The default is at the end of footnotes // Don't worry, if any un cited notes exist, this default value will not be used and it will fall in the upcoming calculation
  let newFootnotePos =
    footnotesElementAndPos.pos + footnotesElementAndPos.node.nodeSize - 1

  footnotesElementAndPos.node.descendants((node, pos, parent, index) => {
    if (isFootnoteNode(node)) {
      // This mainly means that you are iterating over the footnotes to decide which note you want to take place. once you found it, you will insert your note there and the old note (you take place) and every thing after will be shifted by default.
      if (label === ++index) {
        newFootnotePos = footnotesElementAndPos.pos + pos + 1
      }
    }
  })
  return newFootnotePos
}
```

Might need some tweaks.

Reply:
cc: @asouqi